app/scenes/Search/Search.js

```diff
@@ -33,349 +28,352 @@ import DateFilter from "./components/DateFilter";
+  const [pinToTop, setPinToTop] = React.useState(!!match.params.term);
+
+  const locationSearchRef = React.useRef(location.search);
+  const matchParamsTermRef = React.useRef(match.params.term);
+
+  const [results, setResults] = React.useState([]);
+  const [showEmpty] = React.useState(
+    !isLoading && query && results.length === 0
+  );
+  const [showShortcutTip] = React.useState(
+    !pinToTop && location.state && location.state.fromMenu
+  );
+
+  const collectionId = React.useCallback(() => {
+    const id = params.get("collectionId");
     return id ? id : undefined;
-  }
+  }, [params]);
 
-  get userId() {
-    const id = this.params.get("userId");
+  const dateFilter = React.useCallback(() => {
+    const id = params.get("dateFilter");
     return id ? id : undefined;
-  }
+  }, [params]);
 
-  get dateFilter() {
-    const id = this.params.get("dateFilter");
+  const userId = React.useCallback(() => {
+    const id = params.get("userId");
     return id ? id : undefined;
-  }
-
-  get isFiltered() {
-    return (
-      this.dateFilter ||
-      this.userId ||
-      this.collectionId ||
-      this.includeArchived
-    );
-  }
-
-  get title() {
-    const query = this.query;
-    const title = this.props.t("Search");
-    if (query) return `${query} – ${title}`;
-    return title;
-  }
-
-  @action
-  loadMoreResults = async () => {
-    // Don't paginate if there aren't more results or we’re in the middle of fetching
-    if (!this.allowLoadMore || this.isLoading) return;
+  }, [params]);
 
-    // Fetch more results
-    await this.fetchResults();
-  };
+  const includeArchived = React.useCallback(() => {
+    return params.get("includeArchived") === "true";
+  }, [params]);
 
-  @action
-  fetchResults = async () => {
-    if (this.query) {
+  const fetchResults = React.useCallback(async () => {
+    if (query) {
       const params = {
-        offset: this.offset,
+        offset: offset,
         limit: DEFAULT_PAGINATION_LIMIT,
-        dateFilter: this.dateFilter,
-        includeArchived: this.includeArchived,
+        dateFilter: dateFilter(),
+        includeArchived: includeArchived(),
         includeDrafts: true,
-        collectionId: this.collectionId,
-        userId: this.userId,
+        collectionId: collectionId(),
+        userId: userId(),
       };
 
       // we just requested this thing – no need to try again
-      if (this.lastQuery === this.query && isEqual(params, this.lastParams)) {
-        this.isLoading = false;
+      if (lastQuery === query && isEqual(params, lastParams)) {
+        setIsLoading(false);
         return;
       }
 
-      this.isLoading = true;
-      this.lastQuery = this.query;
-      this.lastParams = params;
+      setIsLoading(true);
+      lastQuery.current = query;
+      lastParams.current = params;
 
       try {
-        const results = await this.props.documents.search(this.query, params);
+        setResults(await documents.search(query, params));
```

Comment:
results does not need to be set in state, this might have been the source of your loop 😄 

It's only used to know if we received less than the pagination limit, aka there are more results to load.

Reply:
We are using `results` for `showEmpty` variable and to show all the documents.  We have to use `useState` or `useRef`(if we don't want to re-render when results are updates).

Reply:
The results comes from the documentstore, not the return of the API request directly (for better or worse). Just pinged you on the spot.